src/lev-dashboard/DashboardContent.js

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import styled from 'styled-components';
+import DashboardCounter from "./DashboardCounter";
+
+const CountDiv = styled.div`
+  position: relative;
+  display: flex;
+  flex-direction: row;
+`;
+
+const REFRESH = 10 * 1000;
+const FRAMES = 4;
+const DURATION = 1500;
+
+class DashboardContent extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todaySearches: props.todaySearches,
+      allTime: props.allTime,
+    };
+    this.animateIDs = [];
+  }
+
+  componentDidMount() {
+    this.timerID = setInterval(() => this.updateCounter(), REFRESH);
+  }
+
+  componentWillUnmount() {
+    clearInterval(this.timerID);
+    this.stopAnimation();
+  }
+
+  stopAnimation() {
+    this.animateIDs.forEach(id => clearTimeout(id));
+  }
+
+  animateUpdate(todayOld, todayNew, allOld, allNew) {
+    this.stopAnimation();
+    const todayDiff = todayNew < todayOld ? todayNew : todayNew - todayOld;
+    const allDiff = allNew - allOld;
+    this.setState({
+      todaySearches: todayNew < todayOld ? (todayOld = 0) : todayOld + Math.round(todayDiff / FRAMES),
+      allTime: allOld + Math.round(allDiff / FRAMES)
+    });
+    this.animateIDs = Array.from({ length: FRAMES - 1 },(f, i) =>
```

Comment:
Given that you use `.push` further down, I think this may as well be a for loop that also calls `.push`.

Reply:
I'd even declare some `let` vars as that would simplify the code quite a bit.

Reply:
We need to keep the references to the timeouts for `stopAnimation`, so I'm not sure that would work

Reply:
You'd still be pushing the references on to the array so you would still have them.

Reply:
Will do this as a part of the refactor.